packages/astro/src/runtime/server/index.ts

```diff
@@ -317,6 +317,9 @@ If you're still stuck, please open an issue on GitHub or join us at https://astr
 	}
 
 	if (!hydration) {
+		if (isPage) {
+			return html;
+		}
```

Comment:
For hydration to work, we need to avoid stripping out `astro-fragment` on the top-level page, so we inject a `server:root` prop to track that.